workbench/src/wb_project.c

```diff
@@ -245,14 +248,17 @@ static GSList *wb_project_dir_get_file_list(WB_PROJECT_DIR *root, const gchar *u
 
 	g_hash_table_insert(visited_paths, real_path, GINT_TO_POINTER(1));
 
-	while (TRUE)
+	while ((child_name = g_dir_read_name(dir)))
```

Comment:
I think you should make this more explicit here

Reply:
What do you mean? To write out the code like this:
``` c
	child_name = g_dir_read_name(dir);
	while (child_name)
	{
		children = g_slist_prepend(children, g_strdup(child_name));
		child_name = g_dir_read_name(dir);
	}
```
?

Reply:
I don't think repeating the call is a good idea.  I don't mind the original code myself, but some people prefer things like this: `while ((child_name = g_dir_read_name(dir)) != NULL)` so it's more obvious the assignation is not a typo.

Reply:
At least I needed to think twice what those lines do. I'm not a 100%-C-geek but I don't think we need to use every syntax candy available.

Reply:
Honestly I simply copied techee's code. I also don't like assignment's in while loops that much but I can live with it. I suggest the following code:
``` c
	while ((child_name = g_dir_read_name(dir)) != NULL)
	{
		children = g_slist_prepend(children, g_strdup(child_name));
	}
```
So we have the explicit ```NULL``` pointer check and the curly braces to make it more readable. If that fine for you two then I will apply that change.

Reply:
The original code (with or without the explicit `NULL` check) is pretty much the way I've always seen `readdir` being used. The extra parenthesis aren't required but they make compilers shut up about mistaking `==` for `=`.

Reply:
@codebrainz  So you are suggesting to merge it?

Reply:
Just that the part of the code in question seems reasonable.

Reply:
I changed the code to the last suggested code piece. Please merge.